main.go

```diff
@@ -153,55 +145,97 @@ func (o *Options) LoadConfigurationFile() error {
 	return nil
 }
 
-func run(ctx context.Context) error {
-	// Command line flags
-	var opt Options
+func init() {
+	// init logic
 	opt.InitDefaults()
 
-	if err := opt.LoadConfigurationFile(); err != nil {
-		return fmt.Errorf("loading configuration file: %w", err)
+	f := rootCmd.Flags()
+	f.IntVar(&maxIterations, "max-iterations", 20, "maximum number of iterations agent will try before giving up")
+	f.StringVar(&kubeconfigPath, "kubeconfig-path", "", "path to kubeconfig file")
+	f.StringVar(&promptTemplateFilePath, "prompt-template-file-path", "", "path to custom prompt template file")
+	f.StringVar(&tracePath, "trace-path", "trace.log", "path to the trace file")
+	f.BoolVar(&removeWorkDir, "remove-workdir", false, "remove the temporary working directory after execution")
+
+	// This part is gotten from Options struct? May be put all of them there?
+	f.StringVar(&opt.ProviderID, "llm-provider", opt.ProviderID, "language model provider")
+	f.StringVar(&opt.ModelID, "model", opt.ModelID, "language model e.g. gemini-2.0-flash-thinking-exp-01-21, gemini-2.0-flash")
+	f.BoolVar(&opt.SkipPermissions, "skip-permissions", opt.SkipPermissions, "(dangerous) skip asking for confirmation before executing kubectl commands that modify resources")
+	f.BoolVar(&opt.MCPServer, "mcp-server", opt.MCPServer, "run in MCP server mode")
+	f.BoolVar(&opt.EnableToolUseShim, "enable-tool-use-shim", opt.EnableToolUseShim, "enable tool use shim")
+	f.BoolVar(&opt.Quiet, "quiet", opt.Quiet, "run in non-interactive mode, requires a query to be provided as a positional argument")
+
+	// viper binds and env var prefixes
+	err := viper.BindPFlags(f)
+	if err != nil {
+		print(fmt.Errorf("failed to bind viper flags"))
 	}
+	viper.SetEnvPrefix("KUBECTL_AI")
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	viper.AutomaticEnv()
+}
```

Comment:
I see that `klog.InitFlags(nil)` is no longer here ? (that is useful for controlling the logging configuration).